Solana/src/main/java/com/solana/vendor/borshj/BorshInput.kt

```diff
@@ -46,17 +47,19 @@ interface BorshInput {
             val kotlinParameters = constructor.parameters
 
             val declaredFields = klass.declaredFields
-                .filter { kotlinParameters.map{ kf -> kf.name }.contains(it.name) }
+                .filter { kotlinParameters.map { kf -> kf.name }.contains(it.name) }
+                .sortedBy { field -> field.getAnnotation(FieldOrder::class.java).order }
```

Comment:
I just realized that this doesn't allow for the annotation to be missing; I'm guessing if one of the fields doesn't have the annotation it will crash. The thing is, the field ordering is super important for borsh to work. So I'm not sure if this should be allowed to be optional.

Reply:
I totally got the point. We can ensure they  have e to be there.

Reply:
Added a check in both in & out methods to check for any missing annotations. User-readable message is thrown if missing.